protocol/README.md

```diff
@@ -4,38 +4,49 @@ This document describes the canonical system architecture used to encrypt and de
+  * Any IdP software should work in this scenario provided it can execute some custom code on successful authentication that can:
+    * Read the TDF Client public key from a custom HTTP header sent with the OIDC authentication request.
+    * Construct and send an Attribute Provider web service request, including the public key in the payload.
+    * Return the resulting Claims Object in the signed IdP JWT.
+  * *Virtru Protocol Mapper* is our Keycloak-specific reference implementation of the above functionality.
+* *Attribute Provider* (AP) - A web service that will receive requests from our OIDC IdP custom code (ex: Virtru Protocol Mapper) containing information about authenticated subjet and return our custom OIDC/OAuth claims in response.  It is the responsibility of Attribute Provider to transform 3rd party IdP claims/metadata to ABAC attributes.  It returns a TDF [Claims Object](../schema/ClaimsObject.md).
+* *Key Access Service* (KAS) - Responsible for authorizing and granting TDF Clients access to rewrapped data key material. If authorized, TDF Clients can use this rewrapped data key to decrypt TDF ciphertext.  A valid OIDC token containing [TDF Claims](../schema/ClaimsObject.md) must be used as a bearer token when communicating with KAS.  KAS will first verify the authenticity of the bearer token and then the policy claims within that bearer token.  An otherwise valid and trusted OIDC token without valid TDF Claims will be rejected.
 
 ## Workflow
 
 The following sequence diagrams illustrate the workflow taken by the client to encrypt or decrypt TDF ciphertext. The canonical TDF architecture supports two modes of operation: _online mode_ and _offline mode_, which have distinct workflows as shown below.
 
 _Online mode_ is the default mode, where the [wrapped data key](../schema/KeyAccessObject.md) and [authorization policy](../schema/PolicyObject.md) for TDF ciphertext is committed to KAS in-band as part of the `encrypt` operation. This means that the `encrypt` will succeed if and only if all resources are prepared to facilitate an immediate decrypt.
 
-_Offline mode_ requires that the TDF Client previously authenticated with EAS to get a valid Entity Object. Clients running in this mode commit the [authorization policy](../schema/PolicyObject.md) out-of-band, or when decrypt is first performed. This significantly reduces latency at the cost of larger ciphertext (as the entire wrapped key and policy information must be included). 
-
-Long-lived entity objects can be made available for certain _offline mode_ use cases, for instance, sensors with intermittent or low-bandwidth connectivity can have an Entity Object flashed onto the device during manufacturing.
+_Offline mode_ requires that the TDF Client previously obtained a
+valid, signed, long-lived OIDC JWT with embedded Claims Object via
```

Comment:
```suggestion
long-lived TDF JWT via
```

Reply:
This change was applied (thanks!), so marking this as resolved.